doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,239 @@
+   require modifying the installation ISO and is not suggested, but is
+   supported.
+
+Directly specifying autoinstall as a :code:`autoinstall.yaml` file does not
+require a :code:`#cloud-config` header, and does not use a top level
+:code:`autoinstall` key.  The autoinstall directives are placed at the top
+level.
+
+.. code-block:: yaml
+
+    version: 1
+    ....
+
+
+Cloud-init and Autoinstall interaction
+======================================
+
+While cloud-init may assist in providing the autoinstall configuration to the
+Ubuntu installer, cloud-init itself is not processing the autoinstall.
+
+If cloud-init directives are intended to modify the ephemeral system, they
+must appear at the top level of the cloud-config.  If instead
+cloud-init directives are intended to modify the system being installed, they
+must appear under a :code:`user-data` section in :code:`autoinstall`.
+
+.. code-block:: yaml
+
+    #cloud-config
+    # cloud-init directives may optionally be specified here.
+    # These directives affect the ephemeral system performing the install.
+
+    autoinstall:
+      # autoinstall directives must be specified here, not directly at the top
+      # level.  These directives are processed by the Ubuntu Installer, and
+      # configure the target system to be installed.
+
+      user-data:
+        # cloud-init directives may also be optionally be specified here.
+        # These directives also affect the target system to be installed,
+        # and are processed on first boot.
+
+
+Zero-touch deployment with autoinstall
+======================================
+
+The Ubuntu Installer contains a safeguard, intended to prevent USB Flash Drives
+with an :code:`autoinstall.yaml` file from wiping out the wrong system.
+
+During autoinstall, a prompt will be shown to confirm that the install really
+should proceed and start making modifications to the target system. ::
+
+    start: subiquity/Meta/status_GET
+    Confirmation is required to continue.
+    Add 'autoinstall' to your kernel command line to avoid this
+
+
+    Continue with autoinstall? (yes|no)
+
+To bypass this prompt, arrange for the argument :code:`autoinstall` to be
+present on the kernel command line.
+
+
+Creating an autoinstall config
+==============================
+
+When any system is installed using the Ubuntu installer, an autoinstall file
+for repeating the install is created at
+:code:`/var/log/installer/autoinstall-user-data`.
+
+
+The structure of an autoinstall configuration
+=============================================
+
+See the :ref:`ai` for full details on the supported autoinstall directives.
+
+A minimal autoinstall configuration is:
+
+.. code-block:: yaml
+
+    version: 1
+    identity:
+        hostname: hostname
+        username: username
+        password: $crypted_pass
+
+Here is an example file that shows off most features:
+
+.. parsed-literal::
+
+    :ref:`ai-version`: 1
```

Comment:
I'm curious how this renders, is there some easy way for me to see this locally?

Reply:
`make -C doc install run`

install is setting up the venv.

I'm undecided on this part.  We could just have the standard be that in the sample autoinstalls, we always link to the reference for the top level keys.  If we do blocks as code-formatted yaml, it'll look differently than this part that allows the references.

Reply:
The easiest way to see the preview is with the "build PRs" functionality of readthedocs, but this will need someone with admin permissions on the GitHub repo to set up a webhook to readthedocs. Once this has been done, any PR changes will trigger a preview rebuild which you can access via the "checks" section at the bottom of this page.

Reply:
Thanks so much Sally, the build PRs functionality is now fully connected.  A preview of the docs can be seen with the RTD build check.